src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java

```diff
@@ -77,55 +78,16 @@ protected <T> List<T[]> getChunks(
         return chunks;
     }
 
-    protected void sendBlockHeadersChunks(
-        List<String> blockHeaders,
-        String actualMethod,
-        boolean keepPreviousChunkLastItem
-    ) throws HSMClientException {
+    private void validateBlockHeaders(List<String> blockHeaders, String methodName) throws HSMClientException {
```

Comment:
This method's name seems misleading to me. It is doing more than validating the block headers, which appears to be a pretty straightforward validation. It is also checking the HSM state to know if it can proceed with calling the HSM method. I suggest renaming it to something that suits what it is doing. To give you an idea: `evaluateHsmStateCanBeUpdated`.

Reply:
Makes Sense. Updated